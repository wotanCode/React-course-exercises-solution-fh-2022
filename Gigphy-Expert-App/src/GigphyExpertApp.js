import React, { useState } from 'react'
import { AddCategory } from './components/AddCategory';
import { GifGrid } from './components/GifGrid';

const GigphyExpertApp = ({ defaultCategories = [] }) => {
    
    // const [categories, setCategories] = useState(['One Punch']);
    const [categories, setCategories] = useState( defaultCategories );

    return (
        <>
            <h2>GigphyExpertApp</h2>
            <AddCategory setCategories={ setCategories } />
            <hr />

            <ol>
                {
                    categories.map( category  => (
                        <GifGrid 
                            key={ category }
                            category={ category }
                        />
                    ))
                }
            </ol>

        </>
    )
}

export default GigphyExpertApp;