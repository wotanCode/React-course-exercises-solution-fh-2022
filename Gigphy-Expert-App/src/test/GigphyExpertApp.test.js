import {shallow} from 'enzyme';
import { GigphyExpertApp } from '../GigphyExpertApp';

describe('Pruebas en <GigphyExpertApp />', function() {

  test('Debe de mostrar <GigphyExpertApp /> correctamente', () => {
    const wrapper = shallow(<GigphyExpertApp />);
    expect(wrapper).toMatchSnapshot();
  });

  test('Debe de mostrar una lista de categorias', () => {
    const categories = ['Dragon ball', 'One Piece'];
    const wrapper = shallow(<GigphyExpertApp defaultCategories={categories} />);
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('GifGrid').length).toBe(categories.length);
  });

});