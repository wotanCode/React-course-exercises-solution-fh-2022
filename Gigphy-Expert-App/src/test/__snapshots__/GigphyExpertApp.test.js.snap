// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GigphyExpertApp /> Debe de mostrar <GigphyExpertApp /> correctamente 1`] = `
<Fragment>
  <h2>
    GigphyExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;

exports[`Pruebas en <GigphyExpertApp /> Debe de mostrar una lista de categorias 1`] = `
<Fragment>
  <h2>
    GigphyExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="Dragon ball"
      key="Dragon ball"
    />
    <GifGrid
      category="One Piece"
      key="One Piece"
    />
  </ol>
</Fragment>
`;
